boot/boot.js

```diff
@@ -287,6 +287,92 @@ $tw.utils.decodeURIComponentSafe = function(s) {
 	return v;
 };
 
+/*
+Helpers for encoding/decoding URIs
+*/
+
+/*
+The character that will substitute for a space in the URL
+*/
+var SPACE = "_";
+/*
+The character that will separate out the list elements in the URL
+*/
+var CONJUNCTION = ";";
+/*
+Those of the allowed url characters claimed by TW
+*/
+var claimed= [SPACE, ":", CONJUNCTION];
+/*
+Non-alphanumeric characters allowed in a URL fragment
+More information at https://www.rfc-editor.org/rfc/rfc3986#appendix-A
+*/
+var pchar = "-._~!$&'()*+,;=:@/?";
+/*
+The subset of the pchars we will not percent-encode in permalinks/permaviews
+*/
+var substitutes = pchar.split('').filter(c => claimed.indexOf(c) == -1);
```

Comment:
Arrow functions are not ES5. 
We try to compare with tripple equals whenever possible `.indexOf(c) === -1`
Strings, empty strings or chars use double quotes `""` ... TW should **not have single quotes**, except for eg: `a + '"'` which is probably easier to read than `a + "\""`

Reply:
Damn, I had cleaned out all the ES6+ stuff, then introduced more.  I'll try to fix all this and the issues below this evening.

Thanks for looking.